Data validation function

certara-mtalley opened the pull request:
Internal function to check that covariates exist in the dataset provided. Test updated to incorporate this.

certara-mtalley committed:
Data validation function

Internal function to check that covariates exist in the dataset provided. Test updated to incorporate this.

certara-jcraig commented on R/cov_search.R:
```diff
@@ -13,6 +13,16 @@
+  errors <- data_validation(tab, list_pop_param, cov_continuous, cov_factors)
+  if (length(errors) > 0) {
+    cat("Stopping due to errors detected in data validation:\n")
+    cat(errors, sep = "\n")
```
Let's change this line to `stop(paste0(errors, sep = "\n"), call. = FALSE)` , which will create error condition in R.

certara-jcraig commented on R/cov_search.R:
```diff
@@ -13,6 +13,16 @@
+  if (length(errors) > 0) {
+    cat("Stopping due to errors detected in data validation:\n")
+    cat(errors, sep = "\n")
+    return(invisible())
```
` return(invisible())` can be removed, since the `stop()` error condition will now stop execution.

certara-jcraig replied:
Note that the related test should be changed to: `testthat::expect_error()`

certara-jcraig commented on R/cov_search.R:
```diff
@@ -13,6 +13,16 @@
+    cat(errors, sep = "\n")
+    return(invisible())
+  } else {
+    cat("Data validation successful, selecting covariates...\n")
```
`else` condition can be removed - no need tell the user anything if everything is OK

certara-jcraig reviewed (changes requested):
Thanks @certara-mtalley , just some minor changes requested, see comments.  Let's additionally add this data validation function inside generate_residualsplots and generate_residualsplots2 (I updated ticket with additional scope).

certara-mtalley committed:
Updates per reviewer comments

Updated data validation function. Added it to other functions. Updated tests accordingly.

certara-mtalley committed:
Fixed tests

certara-jcraig reviewed (approved):
Looks good, approving and merging - thanks @certara-mtalley
